DrinkManagerConsole/SearchDrinkConsoleUi.cs

```diff
@@ -21,45 +21,81 @@ public static void StartSearch(List<Drink> drinksList, SearchCriterion searchCri
-                        drinksFound = SearchDrink.SearchByIngredients(new SortedSet<string>(Console.ReadLine()?.Split(' ') ?? throw new InvalidOperationException()), drinksList, searchOption);
+                            var searchOptionChoice = Console.ReadKey();
+                            switch (searchOptionChoice.KeyChar)
+                            {
+                                case '1':
+                                    searchOption = SearchDrinkOption.All;
+                                    incorrectInputIngredientsChoice = false;
+                                    break;
+
+                                case '2':
+                                    searchOption = SearchDrinkOption.Any;
+                                    incorrectInputIngredientsChoice = false;
+                                    break;
+
+                                default:
+                                    Console.WriteLine("\nPlease enter correct answer.");
+                                    incorrectInputIngredientsChoice = true;
+                                    break;
+                            }
+                        } while (incorrectInputIngredientsChoice);
+                        
+                        var ingredientsToSearch = GetTextToSearch(searchCriterion);
+                        drinksFound = SearchDrink.SearchByIngredients(new SortedSet<string>(ingredientsToSearch.Split(' ')), drinksList, searchOption);
                         break;
                 }
 
                 // invoking extracted display method
                 DisplaySearchResults(drinksFound);
 
-                Console.Write($"\nContinue search by {searchCriterion.ToString().ToLower()} (y/n)? ");
-                if (Console.ReadKey().KeyChar == 'n')
+                var incorrectInputEndSearch = true;
+                do
                 {
-                    continueSearch = false;
-                }
+                    Console.Write($"\nContinue search by {searchCriterion.ToString().ToLower()} (yes: [y/enter] / no: [n/esc])? ");
+                    var continueUserInput = Console.ReadKey();
+                    if (continueUserInput.KeyChar == 'y' || continueUserInput.KeyChar == 'Y' || continueUserInput.Key == ConsoleKey.Enter)
+                    {
+                        incorrectInputEndSearch = false;
+                    }
+                    else if (continueUserInput.KeyChar == 'n' || continueUserInput.KeyChar == 'N' || continueUserInput.Key == ConsoleKey.Escape)
+                    {
+                        incorrectInputEndSearch = false;
+                        continueSearch = false;
+                    }
+                } while (incorrectInputEndSearch);
+                
             } while (continueSearch);
         }
 
+        private static string GetTextToSearch(SearchCriterion searchCriterion)
+        {
+            string textToSearch;
+            var incorrectInput = true;
```

Comment:
I'm not sure if this boolean value is needed, maybe we could do string.IsNullOrWhiteSpace(textToSearch) in the while condition ?

Reply:
Nice solution - implemented & pushed.